### PR TITLE
openexr: update to 3.2.1

### PR DESCRIPTION
**Pull request by herbygillot**
###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with ~`sudo port -vst install`~ `sudo port -d destroot`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by mascguy** [comment]
@barracuda156 Sergey, can you test on your legacy systems (PPC, etc)?

**Comment by barracuda156** [comment]
> @barracuda156 Sergey, can you test on your legacy systems (PPC, etc)?

@mascguy Sure, let me try. Will update soon.

**Comment by barracuda156** [comment]
@mascguy @herbygillot Installed fine for me on PPC:
```
36-246% port -v installed openexr
The following ports are currently installed:
  openexr @2.3.0_2 requested_variants='' platform='darwin 10' archs='ppc' date='2022-01-02T13:44:33+0800'
  openexr @3.1.7_1 requested_variants='' platform='darwin 10' archs='ppc' date='2023-05-16T19:34:37+0800'
  openexr @3.1.7_2 requested_variants='' platform='darwin 10' archs='ppc' date='2023-05-17T23:12:09+0800'
  openexr @3.1.7_3 requested_variants='' platform='darwin 10' archs='ppc' date='2023-05-26T21:56:07+0800'
  openexr @3.1.7_4 requested_variants='' platform='darwin 10' archs='ppc' date='2023-06-05T21:54:30+0800'
  openexr @3.1.8_0 requested_variants='' platform='darwin 10' archs='ppc' date='2023-06-09T01:01:11+0800'
  openexr @3.1.9_0 requested_variants='' platform='darwin 10' archs='ppc' date='2023-07-04T17:15:37+0800'
  openexr @3.1.11_0 requested_variants='' platform='darwin 10' archs='ppc' date='2023-08-19T03:29:11+0800'
  openexr @3.2.1_0 (active) requested_variants='' platform='darwin 10' archs='ppc' date='2024-01-26T19:50:14+0800'
```

P. S. It seems to download some stuff during configure. This will likely break builds on old systems (at least on buildbots), unless a dependency on `git` is declared:
```
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Checking whether C compiler has -isysroot
-- Checking whether C compiler has -isysroot - yes
-- Checking whether C compiler supports OSX deployment target flag
-- Checking whether C compiler supports OSX deployment target flag - yes
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/local/bin/gcc-mp-13 - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking whether CXX compiler has -isysroot
-- Checking whether CXX compiler has -isysroot - yes
-- Checking whether CXX compiler supports OSX deployment target flag
-- Checking whether CXX compiler supports OSX deployment target flag - yes
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/local/bin/g++-mp-13 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found PkgConfig: /opt/local/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'libdeflate'
--   Found libdeflate, version 1.19
-- Using libdeflate from /opt/local/lib/libdeflate.dylib
-- Using Imath from /opt/local/lib/cmake/Imath
-- Performing Test OPENEXR_IMF_HAVE_SYSCONF_NPROCESSORS_ONLN
-- Performing Test OPENEXR_IMF_HAVE_SYSCONF_NPROCESSORS_ONLN - Success
-- Performing Test OPENEXR_IMF_HAVE_GCC_INLINE_ASM_AVX
-- Performing Test OPENEXR_IMF_HAVE_GCC_INLINE_ASM_AVX - Failed
-- Looking for include file ucontext.h
-- Looking for include file ucontext.h - not found
-- OpenEXR pkg-config generation enabled
-- Configure OpenEXR 3.2.1, library API version: 31.3.2.1
-- Found Python3: /opt/local/bin/python3.12 (found version "3.12.1") found components: Interpreter 
-- adding bin tests
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/TestImages/GrayRampsHorizontal.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/LuminanceChroma/Garden.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/MultiResolution/ColorCodedLevels.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/MultiResolution/WavyLinesLatLong.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/MultiResolution/WavyLinesCube.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/LuminanceChroma/Flowers.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/MultiView/Adjuster.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/Chromaticities/Rec709_YC.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/Chromaticities/Rec709.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/Chromaticities/XYZ_YC.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/Chromaticities/XYZ.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/TestImages/GammaChart.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/Beachball/singlepart.0001.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/v2/LeftView/Balls.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/v2/Stereo/Trunks.exr
-- Downloading https://raw.githubusercontent.com/AcademySoftwareFoundation/openexr-images/v1.0/Beachball/multipart.0001.exr
-- clang-format not found.
-- Configuring done
-- Generating done
```

**Comment by mascguy** [comment]
Some of the dylib names don't look correct. (Anything with `3_2.31.3.2.1` is questionable.)

Can someone verify whether the previous release had similar dylib naming?

```
a ./opt/local/lib/libIex-3_2.31.dylib
a ./opt/local/lib/pkgconfig
a ./opt/local/lib/libOpenEXRCore-3_2.31.dylib
a ./opt/local/lib/cmake
a ./opt/local/lib/libOpenEXRUtil-3_2.31.3.2.1.dylib
a ./opt/local/lib/libIex-3_2.dylib
a ./opt/local/lib/libIex.dylib
a ./opt/local/lib/libIlmThread-3_2.31.3.2.1.dylib
a ./opt/local/lib/libOpenEXRCore-3_2.31.3.2.1.dylib
a ./opt/local/lib/libOpenEXRCore-3_2.dylib
a ./opt/local/lib/libIlmThread-3_2.dylib
a ./opt/local/lib/libOpenEXR.dylib
a ./opt/local/lib/libOpenEXRCore.dylib
a ./opt/local/lib/libOpenEXR-3_2.31.3.2.1.dylib
a ./opt/local/lib/libOpenEXR-3_2.dylib
a ./opt/local/lib/libOpenEXRUtil.dylib
a ./opt/local/lib/libIlmThread-3_2.31.dylib
a ./opt/local/lib/libOpenEXR-3_2.31.dylib
a ./opt/local/lib/libOpenEXRUtil-3_2.31.dylib
a ./opt/local/lib/libIlmThread.dylib
a ./opt/local/lib/libOpenEXRUtil-3_2.dylib
a ./opt/local/lib/libIex-3_2.31.3.2.1.dylib
```

**Comment by barracuda156** [comment]
@mascguy Judging from https://ports.macports.org/port/openexr/details yes.

File: graphics/openexr/Portfile [hunk 1]
```diff
@@ -42,7 +42,8 @@ if { (!(${universal_possible} && [variant_isset universal]) && ${configure.build
 }
 
 depends_build-append            port:pkgconfig
-depends_lib-append              port:imath \
+depends_lib-append              port:imath      \
+                                port:libdeflate \
                                 port:zlib
```
[the remark this answers is not in the document]
Comment (herbygillot) [reply graphics/openexr/Portfile]: Took at look at all the binaries and libs produced by the build, and none are linking to `zlib`, so I removed it from `depends_lib`.

**Review by mascguy — CHANGES_REQUESTED** [review]

File: graphics/openexr/Portfile [hunk 2]
```diff
@@ -42,8 +42,8 @@ if { (!(${universal_possible} && [variant_isset universal]) && ${configure.build
 }
 
 depends_build-append            port:pkgconfig
-depends_lib-append              port:imath \
-                                port:zlib
+depends_lib-append              port:imath      \
```
[the remark this answers is not in the document]
Comment (herbygillot) [reply graphics/openexr/Portfile]: Done.

**Review by mascguy — APPROVED** [review]
LGTM. Thanks Herby!

**Pull request merged**